### DISCUSSION TITLE
Run command on mode change

### DISCUSSION THREAD

**Discussion by peter-jerry-ye** [helix-editor/helix#8797] [Q&A] [answered]
Hi, I'm using Helix to edit files involving multiple languages, thus I need to switch back my input method to English when I exit the insertion mode so that I can use the normal key bindings. Is there a way to define such scripts when the mode is changed?

Thanks.

**Comment by erasin** [comment, marked as answer]
[read ime-switch config](https://github.com/erasin/helix-config?tab=readme-ov-file#%E8%BE%93%E5%85%A5%E6%B3%95%E5%88%87%E6%8D%A2)

linux/macos use <https://github.com/erasin/helix-config/blob/main/scripts/ime-switch>

windows use <https://github.com/erasin/win-ime-switch>
